spec/models/patient_spec.rb

```diff
@@ -206,5 +206,60 @@
       end
     end
   end
+
+  context '.discard_data' do
+    before do
+      create_list(:prescription_drug, 2, patient: patient)
+      create(:medical_history, patient: patient)
+      create_list(:appointment, 2, patient: patient)
+      bps = create_list(:blood_pressure, 2, patient: patient)
+      sugars = create_list(:blood_sugar, 2, patient: patient)
+      (bps + sugars).each do |record|
+        create(:encounter,
+               :with_observables,
+               patient: patient,
+               observable: record,
+               facility: patient.registration_facility)
+      end
+    end
+
+    it "should discard a patient's address" do
+      patient.discard_data
+      expect(patient.address.discarded?).to be true
+      expect(Address.find_by(id: patient.address.id)).to be nil
+    end
+    it "should discard a patient's appointments" do
+      expect { patient.discard_data }.to change { patient.appointments.count }.by(-2)
+    end
+    it "should discard a patient's blood pressures" do
```

Comment:
We _could_ use `specify` to save us some keystrokes. I think the test code is pretty self-explanatory, doesn't need a description. What do you think?

```ruby
specify { expect { patient.discard_data }.to change { patient.appointments.count }.by(-2) }
```